lib/segment/src/index/query_optimization/condition_converter.rs

```diff
@@ -159,9 +159,34 @@ pub fn field_condition_index<'a>(
         return Some(checker);
     }
 
+    if let Some(checker) = field_condition
+        .geo_polygon
+        .clone()
+        .and_then(|cond| get_geo_polygon_checkers(index, cond))
+    {
+        return Some(checker);
+    }
+
     None
 }
 
+pub fn get_geo_polygon_checkers(
+    index: &FieldIndex,
+    geo_polygon: GeoPolygon,
+) -> Option<ConditionCheckerFn> {
+    match index {
+        FieldIndex::GeoIndex(geo_index) => Some(Box::new(move |point_id: PointOffsetType| {
+            let polygon_wrapper = geo_polygon.convert();
```

Comment:
`polygon_wrapper` should be created outside of the lambda function, so that it will only be created once per query.

Reply:
done.